### ISSUE TITLE
htpasswd generation example does not work

### ISSUE THREAD

**Issue by akantak** [twuni/docker-registry.helm#30]
The example of htpasswd file generation does not work:
```
$ sudo docker run --entrypoint htpasswd registry:2 -Bbn user password > ./htpasswd
docker: Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "htpasswd": executable file not found in $PATH: unknown.
ERRO[0000] error waiting for container: context canceled
```

I tested that with recent image:
```
REPOSITORY                                         TAG                            IMAGE ID       CREATED         SIZE
registry                                           2                              1fd8e1b0bb7e   5 months ago    26.2MB
```

**Comment by Mytholody** [comment]
There appears to be something broken in the latest pull. I believe there is a pull request pending to fix the documentation.
For anyone reading this though...

Try this instead:
`docker run --entrypoint htpasswd registry:2.7.0 -Bbn user password > ./htpasswd`
or
`docker run --rm -ti xmartlabs/htpasswd username password >> ./htpasswd`

**Comment by ross-worth** [comment]
How can a password file be generated on aks where there is no docker access? I've tried generating the file manually but get permission denied

**Comment by Mytholody** [comment]
I'm not familiar with Azure-based setups, but it shouldn't be a problem. You can run it on your local machine and take the key.

Run this with your username and password replaced (and just input the string that is output for your secrets.htpasswd parameter and you should be good to go):
`docker run --entrypoint htpasswd registry:2.7.0 -Bbn user password`

**Comment by ross-worth** [comment]
> I'm not familiar with Azure-based setups, but it shouldn't be a problem. You can run it on your local machine and take the key.
> 
> Run this with your username and password replaced (and just input the string that is output for your secrets.htpasswd parameter and you should be good to go): `docker run --entrypoint htpasswd registry:2.7.0 -Bbn user password`


Thanks Mytholody, that was exactly what I was looking for (which parameter to store the data in)

**Comment by TacoNaco47** [comment]
See: https://stackoverflow.com/questions/62531462/docker-local-registry-exec-htpasswd-executable-file-not-found-in-path/71658782#71658782

`htpasswd` was removed from the latest docker registry images. You can use httpd instead: 
```
 mkdir auth
 docker run \
  --entrypoint htpasswd \
  httpd:2 -Bbn testuser testpassword > auth/htpasswd
```

**Comment by stevek-pro** [comment]
@pieveee How would this work for helm and kubernetes?

**Comment by fguevara-attackiq** [comment]
Just setup a basic auth within the ingress-resource of the chart using annotations and you can now have basic auth. Keep in mind creating the secret in the htpasswd format.

**Comment by wkbrd** [comment]
PR implementing the httpd option is here: https://github.com/twuni/docker-registry.helm/pull/130

**Comment by joshsizer** [comment]
This appears to be fixed with #130, so I am closing the issue.
